packages/web-components/src/menu/menu.ts

```diff
@@ -0,0 +1,327 @@
+   * @public
+   */
+  public setComponent(): void {
+    if (this.$fastController.isConnected && this.slottedMenuList.length && this.slottedTriggers.length) {
+      this._trigger = this.slottedTriggers![0];
+      this._menuList = this.slottedMenuList![0];
+      this._trigger.setAttribute('aria-haspopup', 'true');
+      this._trigger.setAttribute('aria-expanded', `${this.open}`);
+      this.addListeners();
+    }
+  }
+
+  /**
+   * Toggles the open state of the menu.
+   * @public
+   */
+  public toggleMenu = () => {
+    if (this.open) {
+      this.closeMenu();
+    } else {
+      this.openMenu();
+    }
+  };
+
+  /**
+   * Closes the menu.
+   * @public
+   */
+  public closeMenu = () => {
+    this.open = false;
+  };
+
+  /**
+   * Opens the menu.
+   * @public
+   */
+  public openMenu = () => {
+    this.open = true;
+  };
+
+  /**
+   * Focuses on the menu list.
+   * @public
+   */
+  public focusMenuList(): void {
+    if (this.open && this._menuList) {
+      Updates.enqueue(() => {
+        this._menuList!.focus();
+      });
+    }
+  }
+
+  /**
+   * Focuses on the menu trigger.
+   * @public
+   */
+  public focusTrigger(): void {
+    if (!this.open && this._trigger) {
+      Updates.enqueue(() => {
+        this._trigger!.focus();
+      });
+    }
+  }
+
+  /**
+   * Called whenever the open state changes.
+   * Updates the 'aria-expanded' attribute and sets the positioning of the menu.
+   * Sets menu list position
+   * emits openChanged event
+   * @public
+   * @param {boolean} oldValue - The previous value of 'open'.
+   * @param {boolean} newValue - The new value of 'open'.
+   */
+  public openChanged(oldValue: boolean, newValue: boolean): void {
+    if (this.$fastController.isConnected && this._trigger instanceof HTMLElement) {
+      this._trigger.setAttribute('aria-expanded', `${this.open}`);
+      if (this._menuList && this.open) {
+        Updates.enqueue(this.setPositioningTask);
+      }
+    }
+    this.$emit('changed', { open: newValue });
+  }
+
+  /**
+   * Called whenever the 'openOnHover' property changes.
+   * Adds or removes a 'mouseover' event listener to the trigger based on the new value.
+   * @public
+   * @param {boolean} oldValue - The previous value of 'openOnHover'.
+   * @param {boolean} newValue - The new value of 'openOnHover'.
+   */
+  public openOnHoverChanged(oldValue: boolean, newValue: boolean): void {
+    if (newValue) {
+      this._trigger?.addEventListener('mouseover', this.openMenu);
+    } else {
+      this._trigger?.removeEventListener('mouseover', this.openMenu);
+    }
+  }
+
+  /**
+   * The task to set the positioning of the menu.
+   * @protected
+   */
+  protected setPositioningTask = () => {
+    this.setPositioning();
+  };
+
+  /**
+   * Sets the positioning of the menu.
+   * @protected
+   */
+  protected setPositioning(): void {
+    if (this.$fastController.isConnected && this._menuList && this.open && this._trigger) {
+      this.cleanup?.();
```

Comment:
This still keeps the last instance of `autoUpdate` running when the menu is closed, component unmounted...